File: PE_Nonogram/testcase/sol.cpp
#include <bits/stdc++.h>
using namespace std;

void result(const vector<int>& line){
    bool flag = false;
    int cnt = 0;
    for (auto &it: line){
        if (it == 1)
            cnt++;
        else if (cnt > 0){
            cout << cnt << " ";
            cnt = 0;
            flag = true;
        }
    }
    if (cnt > 0 || !flag ) cout << cnt;
    cout << endl;
}

int main() {
    int n;
    cin >> n;
    vector<vector<int>> mp(n, vector<int>(n));
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> mp[i][j];
    for (int j = 0; j < n; j++){
        vector<int> nowCol(n);
        for (int i = 0; i < n; i++)
            nowCol[i] = mp[i][j];
        result(nowCol);
    }
    for (int i = 0; i < n; i++)
        result(mp[i]);
}

File: PE_Nonogram/testcase/sol.py
def result(line):
    cnt = 0
    flag = False
    output = []
    for num in line:
        if num == 1:
            cnt += 1
        elif cnt > 0:
            output.append(str(cnt))
            cnt = 0
            flag = True
    if cnt > 0 or not flag:
        output.append(str(cnt))
    print(" ".join(output))

n = int(input())
matrix = [list(map(int, input().split())) for _ in range(n)]

for j in range(n):
    col = [matrix[i][j] for i in range(n)]
    result(col)

for i in range(n):
    result(matrix[i])
